src/UIKit/UIGestureRecognizer.cs

```diff
@@ -130,12 +132,22 @@ public void RemoveTarget (Token token)
 				throw new ArgumentNullException ("token");
 			if (recognizers == null)
 				return;
-			if (recognizers == token)
-				recognizers = null;
-			Hashtable asHash = recognizers as Hashtable;
-			if (asHash != null)
-				asHash.Remove (token);
-			RemoveTarget (token, token is ParametrizedDispatch ? Selector.GetHandle (parametrized_selector) : Selector.GetHandle (tsel));
+			if (recognizers.ContainsKey (token)){
```

Comment:
Minor code improvement:

```csharp
if (recognizers.TryGetValue (token, out var sel)) {
    recognizers.Remove (token);
    RemoveTarget (token, sel);
}
```

Reply:
Love it

Reply:
Given this targets [modern mono](https://github.com/mono/mono/blob/master/mcs/class/corlib/corlib.dll.sources#L1693) with [corefx APIs](https://github.com/mono/corefx/blob/master/src/Common/src/CoreLib/System/Collections/Generic/Dictionary.cs#L829), can't it just be:
```
if (recognizers.Remove (token, out var sel)) {
    RemoveTarget (token, sel);
}
```